c_check

```diff
@@ -293,6 +293,37 @@ if (($architecture eq "riscv64")) {
     }
 }
 
+$no_lasx = 0;
+if (($architecture eq "loongarch64")) {
+    eval "use File::Temp qw(tempfile)";
+    if ($@){
+        warn "could not load PERL module File::Temp, so could not check compiler compatibility with the LoongArch LASX extension";
+        # Assume no LASX, because currently LASX is only present in vendor
+        # toolchains, undocumented nor upstreamed.
+        $no_lasx = 1;
+    } else {
+        # Old-world assembly flavor: LASX registers named "$xrNN", different
+        # from the ISA manual which suggests "$xNN". This is the flavor we
+        # currently support.
+        #
+        # As the LASX ISA manual is not out yet, we cannot predict what the
+        # new-world flavor would look like, so do not probe flavor for now.
+        # The compiler flags are also unsuitable for new-world gcc.
+        ($fh,$tmpf) = tempfile( SUFFIX => '.c' , UNLINK => 1 );
+        $code = '"xvld $xr0, $a0, 0\n"';
+        print $fh "int main(void){ __asm__ volatile($code); }\n";
+        $args = " -march=loongarch64 -mabi=lp64 -mlasx -c -o $tmpf.o $tmpf";
+        my @cmd = ("$compiler_name $flags $args >/dev/null 2>/dev/null");
+        system(@cmd) == 0;
+        if ($? != 0) {
+            $no_lasx = 1;
+        } else {
+            $no_lasx = 0;
+        }
+        unlink("$tmpf.o");
+    }
+}
+
```

Comment:
Maybe we can use intrinsics here to avoid distinguising '$xNN' or '$xrNN', and so that we don't need to talk about the new world and the old world.

Reply:
You have `.S` files, so these must be taken care of too. C files making use of LASX via intrinsics are unaffected anyway.